### DISCUSSION TITLE
Servant (client) support?

### DISCUSSION THREAD

**Discussion by emberdart** [PostgREST/postgrest#2634] [Q&A]
Hello,

Is it possible to generate client code for `servant` so that `servant-reflex` can take it and use it straight from the browser? This would avoid having to manually define these - having these automated would be amazing. Maybe via https://hackage.haskell.org/package/openapi3-code-generator?

**Comment by laurenceisla** [comment]
Hi, I'm a little lost with `servant` and how it could be used with `postgrest`, could you please elaborate more in how they interact and what you are planning to get?

**Reply by emberdart** [reply]
Sure, so what I expect of PostgREST is that it is a server that gives me access to a database.
One can define a server API using Servant types, whether you're using the server part (which this replaces, essentially, even giving an openapi3 representation, which is amazing) or the client part (which you can use to make requests to other APIs).
So here, what I want to do is to get some kind of code generation to generate the types and client code to make these requests.
I'll try and see what I can do with openapi3-code-generator first, though.

**Reply by laurenceisla** [reply]
I see, I kinda get it now. PostgREST returns an OpenAPI version 2.0 output though, in case you get any trouble working with it.
